behavior/plans/diff.py

```diff
@@ -1,466 +1,288 @@
+    except (PlanDiffUnsupportedParallelException, PlanDiffIncompleteSubinvocationException):
+        # These are not fatal errors. In these cases, we just return None to indicate
+        # that there is no data that needs to be merged.
+        return None
+
+    # Append the diffed numpy array to diffed_matrixes for us to post-process at once.
+    diffed_matrixes.append(matrix)
+    return None
+
+
+def separate_subinvocation(start_times, end_times, root_start_times, root_end_times, subinvocations):
+    """
+    For a given query session template, this function identifies the OUs associated with a
+    given invocation of the query template.
 
-        # Create compound key.
-        df["statement_id"] = (
-            df["query_id"].astype(str) + "_" + df["statement_timestamp"].astype(str) + "_" + df["pid"].astype(str)
-        )
-        remapped[ou_name] = df
-
-    return remapped
-
-
-def infer_query_plans(unified, logdir):
-    # Count all query plan nodes
-    query_to_plan_node_counts = {}
-    for query_id in pd.unique(unified.index):
-        node_ids = unified.loc[[query_id]]["plan_node_id"]
-        plan_node_counts = node_ids.value_counts().to_dict()
-        query_to_plan_node_counts[query_id] = plan_node_counts
-
-    # If less than 1% of plans have the last node id, just drop it.
-    inferred_plan_node_counts = {}
-    inferred_query_plans = {}
-
-    for query_id, node_id_to_count in query_to_plan_node_counts.items():
-        max_count = max(node_id_to_count.values())
-        truncation_node_id = None
-
-        for (node_id, count) in node_id_to_count.items():
-            if count < (0.01 * max_count):
-                logger.warning(
-                    "Truncating potentially broken plan.  Query_id: %s | Truncation Node Id: %s | Node Id to Count: %s",
-                    query_id,
-                    truncation_node_id,
-                    sorted(node_id_to_count.items()),
-                )
-                truncation_node_id = node_id
-                break
-
-        # Only truncate plans if the criteria is met.
-        if truncation_node_id is not None:
-            inferred_plan_node_counts[query_id] = {k: v for k, v in node_id_to_count.items() if k < truncation_node_id}
-        else:
-            inferred_plan_node_counts[query_id] = node_id_to_count
-
-    with (logdir / "inferred_plan_node_counts.json").open("w", encoding="utf-8") as f:
-        json_dict = {
-            str(query_id): {int(node_id): int(node_count) for node_id, node_count in sorted(plan_node_counter.items())}
-            for query_id, plan_node_counter in inferred_plan_node_counts.items()
-        }
-        json.dump(json_dict, f, indent=4)
-
-    inferred_query_plans = {
-        query_id: set(plan_node_id_to_counts.keys())
-        for query_id, plan_node_id_to_counts in inferred_plan_node_counts.items()
-    }
+    Parameters
+    ----------
+    start_times : numpy.array[int64]
```

Comment:
```suggestion
    start_times : np.array[int64]
```
Since you only import as np